--- src/sat/glucose2/Glucose2.cpp.orig	2020-12-25 15:00:29.119838318 +0200
+++ src/sat/glucose2/Glucose2.cpp	2020-12-25 15:02:26.599484344 +0200
@@ -35,6 +35,8 @@
 
 #include "sat/glucose2/CGlucose.h"
 
+#include "misc/util/abc_global.h"
+
 ABC_NAMESPACE_IMPL_START
 
 using namespace Gluco2;
--- src/sat/glucose2/ParseUtils.h.orig	2020-12-25 14:50:53.431294262 +0200
+++ src/sat/glucose2/ParseUtils.h	2020-12-25 14:51:01.078207347 +0200
@@ -25,7 +25,7 @@
 #include <stdio.h>
 #include <math.h>
 
-#include "misc/zlib/zlib.h"
+#include "zlib.h"
 
 ABC_NAMESPACE_CXX_HEADER_START
 
