diff --git a/Makefile b/Makefile
index b583f7d..e7868f7 100644
--- a/Makefile
+++ b/Makefile
@@ -39,7 +39,7 @@ MODULES := \
 	src/map/mapper src/map/mio src/map/super src/map/if src/map/if/acd \
 	src/map/amap src/map/cov src/map/scl src/map/mpm src/map/emap \
 	src/misc/extra src/misc/mvc src/misc/st src/misc/util src/misc/nm \
-	src/misc/vec src/misc/hash src/misc/tim src/misc/bzlib src/misc/zlib \
+	src/misc/vec src/misc/hash src/misc/tim \
 	src/misc/mem src/misc/bar src/misc/bbl src/misc/parse src/misc/btor \
 	src/opt/cut src/opt/fxu src/opt/fxch src/opt/rwr src/opt/mfs src/opt/sim \
 	src/opt/ret src/opt/fret src/opt/res src/opt/lpk src/opt/nwk src/opt/rwt src/opt/rar \
@@ -78,6 +78,7 @@ $(ARCHFLAGS_EXE) : arch_flags.c
 	$(CC) $< -o $(ARCHFLAGS_EXE)
 
 INCLUDES += -I$(ABCSRC)/src
+LIBS=-lbz2 -lz
 
 # Use C99 stdint.h header for platform-dependent types
 ifdef ABC_USE_STDINT_H
diff --git a/src/base/io/ioReadAiger.c b/src/base/io/ioReadAiger.c
index 200247b..af12c32 100644
--- a/src/base/io/ioReadAiger.c
+++ b/src/base/io/ioReadAiger.c
@@ -26,8 +26,8 @@
 #include <string.h>
 #include <assert.h>
 
-#include "misc/bzlib/bzlib.h"
-#include "misc/zlib/zlib.h"
+#include <bzlib.h>
+#include <zlib.h>
 #include "ioAbc.h"
 
 ABC_NAMESPACE_IMPL_START
diff --git a/src/base/io/ioReadBlifMv.c b/src/base/io/ioReadBlifMv.c
index 05fc5d2..afbb237 100644
--- a/src/base/io/ioReadBlifMv.c
+++ b/src/base/io/ioReadBlifMv.c
@@ -18,8 +18,8 @@
 
 ***********************************************************************/
 
-#include "misc/zlib/zlib.h"
-#include "misc/bzlib/bzlib.h"
+#include <zlib.h>
+#include <bzlib.h>
 #include "base/abc/abc.h"
 #include "misc/vec/vecPtr.h"
 #include "ioAbc.h"
diff --git a/src/base/io/ioWriteAiger.c b/src/base/io/ioWriteAiger.c
index cad335d..110256a 100644
--- a/src/base/io/ioWriteAiger.c
+++ b/src/base/io/ioWriteAiger.c
@@ -26,8 +26,8 @@
 #include <string.h>
 #include <assert.h>
 
-#include "misc/bzlib/bzlib.h"
-#include "misc/zlib/zlib.h"
+#include <bzlib.h>
+#include <zlib.h>
 #include "ioAbc.h"
 
 
diff --git a/src/map/scl/sclLiberty.c b/src/map/scl/sclLiberty.c
index 2c3f1b7..e519fbe 100644
--- a/src/map/scl/sclLiberty.c
+++ b/src/map/scl/sclLiberty.c
@@ -25,7 +25,7 @@
 #include <fnmatch.h>
 #endif
 
-#include "misc/zlib/zlib.h"
+#include <zlib.h>
 #include "sclLib.h"
 #include "misc/st/st.h"
 #include "map/mio/mio.h"
diff --git a/src/sat/bsat2/MainSat.cpp b/src/sat/bsat2/MainSat.cpp
index 8554662..d1c2eca 100644
--- a/src/sat/bsat2/MainSat.cpp
+++ b/src/sat/bsat2/MainSat.cpp
@@ -21,7 +21,7 @@ OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWA
 #include <errno.h>
 
 #include <signal.h>
-#include "misc/zlib/zlib.h"
+#include <zlib.h>
 
 #include "sat/bsat2/System.h"
 #include "sat/bsat2/ParseUtils.h"
diff --git a/src/sat/bsat2/MainSimp.cpp b/src/sat/bsat2/MainSimp.cpp
index 2d0ad10..40ce973 100644
--- a/src/sat/bsat2/MainSimp.cpp
+++ b/src/sat/bsat2/MainSimp.cpp
@@ -21,7 +21,7 @@ OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWA
 #include <errno.h>
 
 #include <signal.h>
-#include "misc/zlib/zlib.h"
+#include <zlib.h>
 
 #ifndef _WIN32
 #include <sys/resource.h>
diff --git a/src/sat/bsat2/ParseUtils.h b/src/sat/bsat2/ParseUtils.h
index b7325f1..8ea8d49 100644
--- a/src/sat/bsat2/ParseUtils.h
+++ b/src/sat/bsat2/ParseUtils.h
@@ -24,7 +24,7 @@ OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWA
 #include <stdlib.h>
 #include <stdio.h>
 
-#include "misc/zlib/zlib.h"
+#include <zlib.h>
 
 ABC_NAMESPACE_CXX_HEADER_START
 
diff --git a/src/sat/cnf/cnfMan.c b/src/sat/cnf/cnfMan.c
index cfcc437..96dff25 100644
--- a/src/sat/cnf/cnfMan.c
+++ b/src/sat/cnf/cnfMan.c
@@ -21,7 +21,7 @@
 #include "cnf.h"
 #include "sat/bsat/satSolver.h"
 #include "sat/bsat/satSolver2.h"
-#include "misc/zlib/zlib.h"
+#include <zlib.h>
 
 ABC_NAMESPACE_IMPL_START
 
diff --git a/src/sat/glucose/ParseUtils.h b/src/sat/glucose/ParseUtils.h
index a3f25a6..eb5cd06 100644
--- a/src/sat/glucose/ParseUtils.h
+++ b/src/sat/glucose/ParseUtils.h
@@ -25,7 +25,7 @@ OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWA
 #include <stdio.h>
 #include <math.h>
 
-#include "misc/zlib/zlib.h"
+#include <zlib.h>
 
 ABC_NAMESPACE_CXX_HEADER_START
 
