Index: gpasm/directive.c
===================================================================
--- gpasm/directive.c	(revision 1330)
+++ gpasm/directive.c	(working copy)
@@ -128,6 +128,19 @@
           --Max_size;
           break;
 
+	  // special characters
+	case '0': //(NULL)
+	case 'a': //(bell)
+	case 'b': //(backspace)
+	case 't': //(tab)
+	case 'n': //(newline)
+	case 'v': //(vertical tab)
+	case 'f': //(form feed)
+	case 'r': //(return)
+	  *d++ = '\\';
+	  *d++ = ch;
+	  break;
+
         default: {
           *d++ = ch;
           --Max_size;
