--- python3-cocotb-1.8.1/cocotb/share/makefiles/simulators/Makefile.vcs.orig
+++ python3-cocotb-1.8.1/cocotb/share/makefiles/simulators/Makefile.vcs
@@ -71,13 +71,13 @@ endif
 # Can't do this using an argument, we have to create a PLI table file
 # enabling write access to the design
 $(SIM_BUILD)/pli.tab : | $(SIM_BUILD)
-	echo "acc+=rw,wn:*" > $@
+	echo "acc+=rw,wn,cbk:*" > $@
 
 # Compilation phase
 $(SIM_BUILD)/simv: $(VERILOG_SOURCES) $(SIM_BUILD)/pli.tab $(CUSTOM_COMPILE_DEPS) | $(SIM_BUILD)
 	cd $(SIM_BUILD) && \
 	TOPLEVEL=$(TOPLEVEL) \
-	$(CMD) -top $(TOPLEVEL) $(PLUSARGS) +acc+1 +vpi -P pli.tab +define+COCOTB_SIM=1 -sverilog \
+	$(CMD) -top $(TOPLEVEL) $(PLUSARGS) -debug_access+all -debug_region+cell+encrypt -debug_region+cell+lib +vpi -P pli.tab +define+COCOTB_SIM=1 -sverilog \
 	-timescale=$(COCOTB_HDL_TIMEUNIT)/$(COCOTB_HDL_TIMEPRECISION) \
 	$(EXTRA_ARGS) -debug -load $(shell cocotb-config --lib-name-path vpi vcs) $(COMPILE_ARGS) $(VERILOG_SOURCES)
 
-- 
2.45.0

